### PR TITLE
Updating index page with journey details

### PR DESCRIPTION
**Pull request by oraNod**
This PR takes details from the journey maps here: https://github.com/ansible/docsite/blob/personas/user-journeys/ansible-user-journey-maps.md

And adds those details to the index page. Note that we decided to remove the section for basic community member from the index page. We'll have a general "Join the community" page for those details.

### CODE REVIEW

File: data/pages.yaml [hunk 1]
```diff
@@ -19,13 +21,28 @@ index:
...
+    action_label: Button group that links to base Ansible developer documentation.
+    action_one: Set up your development environment
+    action_two: Review the Ansible contributor guidelines
+    action_three: Write custom modules or plugins
+  developer_resources_heading: Developer resources
+  developer_resources_body: Contribute to an Ansible project or collection, run sanity tests, write integration and unit tests, and more.
+  developer_resources_action: Find developer resources
+  #Community maintainer
+  maintainer_heading: Community maintainers
+  maintainer_base_body: Ansible maintainers are responsible for the ongoing quality of collections and play a vital role in guiding the direction of the wider community.
```
Comment (samccann) [inline data/pages.yaml:39]:
```suggestion
  maintainer_base_body: Ansible maintainers are responsible for the ongoing quality of  Ansible projects and collections and play a vital role in guiding the direction of the wider community.
```

File: data/pages.yaml [hunk 2]
```diff
@@ -19,13 +21,28 @@ index:
...
+  maintainer_base_actions:
+    #Labels provide meaningful descriptions of elements for accessibility purposes.
+    action_label: Button group that links to base Ansible community maintainer documentation.
+    action_one: Learn about the Ansible community
+    action_two: Review and agree to the code of conduct
+    action_three: Join or start a discussion that interests you
+  maintainer_resources_heading: Community maintainer resources
+  maintainer_resources_body: Handle project lifecycles, ensure the health of collections, join the Ansible steering committe, and more.
```
Comment (samccann) [inline data/pages.yaml:47]:
```suggestion
  maintainer_resources_body: Handle project lifecycles, ensure the health of collections, join the Ansible steering committee, and more.
```

File: data/pages.yaml [hunk 3]
```diff
@@ -19,13 +21,28 @@ index:
...
-  dev_body:
-  dev_secondary_heading:
-  dev_secondary_body:
-  dev_button:
-    button_one:
-    button_two:
-    button_three:
\ No newline at end of file
+  developer_heading: Ansible developers
+  developer_base_body: Ansible is open-source. You can add code that fixes bugs, implements new features, or extends the benefits of simple, agentless automation.
```
Comment (samccann) [inline data/pages.yaml:27]:
```suggestion
  developer_base_body: Ansible is open source. You can add code that fixes bugs, implements new features, or extends the benefits of simple, agentless automation.
```

**Review by samccann — COMMENTED** [review]
a few nits. I think we need to change some of the links but that can come in a different PR.

**Comment by oraNod** [comment]
there's an issue for the links: https://github.com/ansible/jinja-docsite/issues/9

**Pull request merged**